pkg/graveler/delete_sensor.go

```diff
@@ -0,0 +1,154 @@
+package graveler
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/treeverse/lakefs/pkg/logging"
+)
+
+const (
+	callbackChannelSize = 1000
+	graceDuration       = 30 * time.Second
+)
+
+type DeleteSensorCB func(repositoryID RepositoryID, branchID BranchID, stagingTokenID StagingToken, inGrace bool)
+
+type StagingTokenCounter struct {
+	StagingTokenID StagingToken
+	Counter        int
+}
+
+type stagingTokenData struct {
+	repositoryID   RepositoryID
+	branchID       BranchID
+	stagingTokenID StagingToken
+}
+
+func (s *stagingTokenData) CombinedKey() string {
```

Comment:
Please document this function.  Why do we need it?  Go doesn't need a string key to index a map, so if it's just for that -- you can use a struct of strings as a [map key](https://go.dev/blog/maps#key-types) _and it will work_.

Reply:
Didn't want to add another level to check and validate, documenting